secboot/secboot_tpm_test.go

```diff
@@ -406,103 +406,120 @@ func (s *secbootSuite) TestUnlockIfEncrypted(c *C) {
 
 func (s *secbootSuite) TestSealKey(c *C) {
 	tmpDir := c.MkDir()
+	mockErr := errors.New("some error")
 
-	var mockEFI []string
-	for _, name := range []string{"a", "b", "c", "d", "e", "f"} {
-		mockFileName := filepath.Join(tmpDir, name)
-		err := ioutil.WriteFile(mockFileName, nil, 0644)
-		c.Assert(err, IsNil)
-		mockEFI = append(mockEFI, mockFileName)
-	}
+	for _, tc := range []struct {
+		tpmErr               error
+		tpmEnabled           bool
+		missingFile          bool
+		addEFISbPolicyErr    error
+		addSystemdEFIStubErr error
+		addSnapModelErr      error
+		provisioningErr      error
+		sealErr              error
+		provisioningCalls    int
+		sealCalls            int
+		expectedErr          string
+	}{
+		{tpmErr: mockErr, expectedErr: "cannot connect to TPM: some error"},
+		{tpmEnabled: false, expectedErr: "TPM device is not enabled"},
+		{tpmEnabled: true, missingFile: true, expectedErr: "file /does/not/exist does not exist"},
+		{tpmEnabled: true, addEFISbPolicyErr: mockErr, expectedErr: "cannot add EFI secure boot policy profile: some error"},
+		{tpmEnabled: true, addSystemdEFIStubErr: mockErr, expectedErr: "cannot add systemd EFI stub profile: some error"},
+		{tpmEnabled: true, addSnapModelErr: mockErr, expectedErr: "cannot add snap model profile: some error"},
+		{tpmEnabled: true, provisioningErr: mockErr, provisioningCalls: 1, expectedErr: "cannot provision TPM: some error"},
+		{tpmEnabled: true, sealErr: mockErr, provisioningCalls: 1, sealCalls: 1, expectedErr: "some error"},
+		{tpmEnabled: true, provisioningCalls: 1, sealCalls: 1, expectedErr: ""},
+	} {
+		var mockEFI []string
+		for _, name := range []string{"a", "b", "c", "d", "e", "f"} {
```

Comment:
you probably want to have each test case use a new tmp directory, no? if so you need something like this:

```suggestion
		tmpDir := c.MkDir()
		for _, name := range []string{"a", "b", "c", "d", "e", "f"} {
```

as-is, the first test case will create all these files and then every test afterwards will re-create the same files with the same content again. 

If you want every test case to have this same set of files, you're probably better off leaving this outside the loop